diff --git a/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he.dts b/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he.dts
index f132286b355..13d8a70f863 100644
--- a/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he.dts
+++ b/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he.dts
@@ -61,6 +61,7 @@
 		zephyr,sram = &dtcm;
 		zephyr,console = &uart4;
 		zephyr,shell-uart = &uart4;
+		zephyr,bt-uart = &uart0;
 	};

 	aliases {
@@ -75,6 +76,10 @@
 	};
 };

+&uart0 {
+	pinctrl-0 = < &pinctrl_uart0 >;
+	status = "okay";
+};
 &uart4 {
 	status = "okay";
 };
diff --git a/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he_defconfig b/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he_defconfig
index 9a72771efb8..78bf6f21cf4 100644
--- a/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he_defconfig
+++ b/boards/arm/alif_e7_devkit/alif_e7_dk_rtss_he_defconfig
@@ -32,4 +32,4 @@ CONFIG_BUILD_NO_GAP_FILL=y

 CONFIG_UART_NS16550=y
 # UART variant of ns16550 that supports auto flow control
-CONFIG_UART_NS16550_VARIANT_NS16750=y
+CONFIG_UART_NS16550_VARIANT_NS16750=n
diff --git a/dts/arm/alif/ensemble-pinctrl.dtsi b/dts/arm/alif/ensemble-pinctrl.dtsi
index 8acb7e7f0fb..b9b9ab06e7b 100644
--- a/dts/arm/alif/ensemble-pinctrl.dtsi
+++ b/dts/arm/alif/ensemble-pinctrl.dtsi
@@ -16,6 +16,12 @@
 };

 &pinctrl {
+	pinctrl_uart0: pinctrl_uart0 {
+		group0 {
+			pinmux = < PIN_P1_4__UART0_RX_B >,
+				 < PIN_P1_5__UART0_TX_B >;
+		};
+	};
 	pinctrl_uart4: pinctrl_uart4 {
 		group0 {
 			pinmux = < PIN_P12_1__UART4_RX_B >,
diff --git a/dts/arm/alif/ensemble_rtss_common.dtsi b/dts/arm/alif/ensemble_rtss_common.dtsi
index 344c04e0935..6eb6cd59d9e 100644
--- a/dts/arm/alif/ensemble_rtss_common.dtsi
+++ b/dts/arm/alif/ensemble_rtss_common.dtsi
@@ -124,6 +124,19 @@
 		reg = <0x4902E0F0 0x10>;
 		interrupts = < 120 3 >;
 	};
+
+	uart0: uart@49018000 {
+		compatible = "ns16550";
+		reg = <0x49018000 0x100>;
+		/* 100Mhz baud clock */
+		clock-frequency = <100000000>;
+		interrupts = <124 3>;
+		reg-shift = <2>;
+		current-speed = <115200>;
+		pinctrl-names = "default";
+		status = "disabled";
+	};
+
 	uart2: uart@4901a000 {
 		compatible = "ns16550";
 		reg = <0x4901a000 0x100>;
diff --git a/samples/bluetooth/beacon/prj.conf b/samples/bluetooth/beacon/prj.conf
index 1d6745c7942..66c2d16a006 100644
--- a/samples/bluetooth/beacon/prj.conf
+++ b/samples/bluetooth/beacon/prj.conf
@@ -1,3 +1,3 @@
 CONFIG_BT=y
 CONFIG_BT_DEBUG_LOG=y
-CONFIG_BT_DEVICE_NAME="Test beacon"
+CONFIG_BT_DEVICE_NAME="FANSTEL BT40F"
